src/header.rs

```diff
@@ -0,0 +1,243 @@
+use error::Mp3Error;
+use tables::*;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Version {
+    Mpeg2_5,
+    Reserved,
+    Mpeg2,
+    Mpeg1,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Layer {
+    LayerI,
+    LayerII,
+    LayerIII,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Protection {
+    Yes,
+    No,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Private {
+    Yes,
+    No,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Padding {
+    Yes,
+    No,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Mode {
+    Stereo,
+    JointStereo,
+    DualChannel,
+    Mono,
+}
+
+/*
+|------|----------------|------------------------------|
+|      | Layer I and II |          Layer III           |
+|------|----------------|------------------|-----------|
+| bits |  Layer I & II  | Intensity Stereo | MS Stereo |
+|------|----------------|------------------|-----------|
+|  00  | bands 4 to 31  |        Off       |    Off    |
+|  01  | bands 8 to 31  |        On        |    Off    |
+|  10  | bands 12 to 31 |        Off       |    On     |
+|  11  | bands 16 to 31 |        On        |    On     |
+|------|----------------|------------------|-----------|
+*/
+#[derive(Debug, Clone, PartialEq)]
+pub enum ModeExtension {
+    Bands(u8),
+    Stereo(bool, bool),
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Copyright {
+    Yes,
+    No,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Original {
+    Yes,
+    No,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Emphasis {
+    None,
+    Ms50_15,  // 50/15 ms
+    Reserved,
+    CcittJ17,  // CCITT J.17
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct FrameHeader {
+    version: Version,
+    layer: Layer,
+    protection: Protection,
+    bitrate: u16, // kbps
+    sampling_rate: u16, // Hz
+    padding: Padding,
+    private: Private,
+    mode: Mode,
+    mode_extension: ModeExtension,
+    copyright: Copyright,
+    original: Original,
+    emphasis: Emphasis,
+}
+
+
+pub fn frame_header(data: &[u8]) -> Result<FrameHeader, Mp3Error> {
```

Comment:
Maybe this should be a `TryFrom<&[u8]> for FrameHeader`?

Reply:
`TryFrom` is nightly only. Also, it'd hide the function amongst other trait impls, which isn't good.

IMO the function should be renamed though, to something more descriptive `parse_frame_header`.